staging/src/k8s.io/client-go/dynamic/fake/simple_test.go

```diff
@@ -446,6 +455,396 @@ func TestListWithTypedFixtures(t *testing.T) {
+			ns:       "ns-foo",
+			createOptions: metav1.CreateOptions{
+				DryRun: []string{metav1.DryRunAll},
+			},
+			wantActions: []clientgotesting.Action{
+				clientgotesting.CreateActionImpl{
+					ActionImpl: clientgotesting.ActionImpl{
+						Namespace: "ns-foo",
+						Verb:      "create",
+						Resource: schema.GroupVersionResource{
+							Group:    "group",
+							Version:  "version",
+							Resource: "thekinds",
+						},
+					},
+					CreateOptions: metav1.CreateOptions{
+						DryRun: []string{metav1.DryRunAll},
+					},
+					Object: newUnstructured("group/version", "TheKind", "ns-foo", "name-foo"),
+				},
+			},
+		},
+		"resource creation with no namespace": {
+			resource: newUnstructuredWithNoNamespace("group/version", "TheKind", "name-foo"),
+			createOptions: metav1.CreateOptions{
+				DryRun: []string{metav1.DryRunAll},
+			},
+			wantActions: []clientgotesting.Action{
+				clientgotesting.CreateActionImpl{
+					ActionImpl: clientgotesting.ActionImpl{
+						Verb: "create",
+						Resource: schema.GroupVersionResource{
+							Group:    "group",
+							Version:  "version",
+							Resource: "thekinds",
+						},
+					},
+					CreateOptions: metav1.CreateOptions{
+						DryRun: []string{metav1.DryRunAll},
+					},
+					Object: newUnstructuredWithNoNamespace("group/version", "TheKind", "name-foo"),
+				},
+			},
+		},
+		"subresource creation with namespace": {
+			ns:           "ns-foo",
+			subresources: []string{"status"},
+			createOptions: metav1.CreateOptions{
+				FieldManager: "test",
+			},
+			resource: newUnstructured("group/version", "TheKind", "ns-foo", "name-foo"),
+			wantActions: []clientgotesting.Action{
+				clientgotesting.CreateActionImpl{
+					Name: "name-foo",
+					ActionImpl: clientgotesting.ActionImpl{
+						Namespace:   "ns-foo",
+						Verb:        "create",
+						Subresource: "status",
+						Resource: schema.GroupVersionResource{
+							Group:    "group",
+							Version:  "version",
+							Resource: "thekinds",
+						},
+					},
+					CreateOptions: metav1.CreateOptions{
+						FieldManager: "test",
+					},
+					Object: newUnstructured("group/version", "TheKind", "ns-foo", "name-foo"),
+				},
+			},
+		},
+		"subresource creation with no namespace": {
+			subresources: []string{"status"},
+			createOptions: metav1.CreateOptions{
+				FieldManager: "test",
+			},
+			resource: newUnstructuredWithNoNamespace("group/version", "TheKind", "name-foo"),
+			wantActions: []clientgotesting.Action{
+				clientgotesting.CreateActionImpl{
+					Name: "name-foo",
+					ActionImpl: clientgotesting.ActionImpl{
+						Verb:        "create",
+						Subresource: "status",
+						Resource: schema.GroupVersionResource{
+							Group:    "group",
+							Version:  "version",
+							Resource: "thekinds",
+						},
+					},
+					CreateOptions: metav1.CreateOptions{
+						FieldManager: "test",
+					},
+					Object: newUnstructuredWithNoNamespace("group/version", "TheKind", "name-foo"),
+				},
```

Comment:
For completeness can we also add a test case to TestCreate for empty CreateOptions?